pkg/base1/cockpit.js

```diff
@@ -1379,7 +1387,9 @@ function basic_scope(cockpit, jquery) {
 
         var ret;
 
-        if (suffix)
+        if (string_representation === "")
+            ret = [];
```

Comment:
What about skipping this case and just returning `[ "" ]` or ´[ "", suffix ]`?  Otherwise this is a API break.